components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/configuration/models/NewVersionNotifierConfiguration.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.core.configuration.models;
+
+import org.wso2.carbon.kernel.annotations.Configuration;
+import org.wso2.carbon.kernel.annotations.Element;
+
+
+/**
+ * Class to hold New API Version created Notification parameters.
+ */
+@Configuration(description = "API Version Notifier Configurations")
+public class NewVersionNotifierConfiguration {
```

Comment:
we discussed to remove notification type from deploymnt yaml  it s not useful to put there. because users only can change the notifiers not types.so users can add own notifiers here.

![deployment](https://user-images.githubusercontent.com/8374895/29940346-8c896a1a-8eac-11e7-9db1-ca74add307fc.png)

Reply:
For the notifier template can't we use arrays for different notifiers? Element 1 for email and element 2 for sms. Then we can have a property map (Hashmap in yaml context) to include classs, template, title. This allows users to define multiple properties for their custom impl.